api/rest/coordinator_test.go

```diff
@@ -0,0 +1,143 @@
+package rest
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCoordinator_CreatorShouldCreateAbundleAndReturnUpdateChan(t *testing.T) {
+
+	numGoroutine := runtime.NumGoroutine()
+
+	client := new(MockClient)
+
+	c := ParallelCoordinator{
+		client: client,
+	}
+
+	ctx := context.TODO()
+
+	node1 := node{IP: net.ParseIP("192.0.2.1"), baseURL: "http://192.0.2.1"}
+	node2 := node{IP: net.ParseIP("192.0.2.2"), baseURL: "http://192.0.2.2"}
+	node3 := node{IP: net.ParseIP("192.0.2.3"), baseURL: "http://192.0.2.3"}
+	testNodes := []node{
+		node1,
+		node2,
+		node3,
+	}
+
+	expected := []BundleStatus{}
+
+	for _, n := range testNodes {
+		id := fmt.Sprintf("%s-%s", n.IP, "id")
+		client.On("CreateBundle", ctx, n.baseURL, id).Return(&Bundle{ID: id, Status: Started}, nil)
+		client.On("Status", ctx, n.baseURL, id).Return(&Bundle{ID: id, Status: Done}, nil)
+
+		expected = append(expected,
+			BundleStatus{ID: id, node: n},
+			BundleStatus{ID: id, node: n, done: true},
+		)
+	}
+	s := c.Create(context.TODO(), "id", testNodes)
+
+	var statuses []BundleStatus
+
+	assert.Equal(t, numberOfWorkers, runtime.NumGoroutine()-numGoroutine)
```

Comment:
I saw flaky failures due to this assert and I don't think it's going to be entirely reliable to check that the number of additional goroutines is some exact amount.